lib/ts/hugepages.h

```diff
@@ -20,13 +20,26 @@
  */
 #ifndef _hugepages_h_
 #define _hugepages_h_
+#include "ink_memory.h"
 
 #include <cstring>
 
+#define HUGEPAGE_SIZE  (PAGE_SIZE<<9)
+
 size_t ats_hugepage_size(void);
 bool ats_hugepage_enabled(void);
 void ats_hugepage_init(int);
 void *ats_alloc_hugepage(size_t);
 bool ats_free_hugepage(void *, size_t);
+void *ats_alloc_hugepage_stack(size_t);
+
+void *huge_normal_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+void *huge_nodump_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+bool huge_dalloc(void *chunk, size_t size, bool committed, unsigned arena_ind);
+bool huge_commit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_decommit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_purge(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_split(void *chunk, size_t size, size_t size_a, size_t size_b, bool committed, unsigned arena_ind);
+bool huge_merge(void *chunk_a, size_t size_a, void *chunk_b, size_t size_b, bool committed, unsigned arena_ind);
```

Comment:
This is over-allocating the pages so that they are (1) aligned on a hugepage boundary and (2) oversized in allocation so that we don't need to commit/decommit individual small-pages.

This is the way we get zero-latency hugepages.  It works nicely.  :->